src/renderer/services/wallets.js

```diff
@@ -73,31 +79,168 @@ export default {
   },
 
   openSavedWallet(name, passphrase) {
-    return new Promise((resolve, reject) => {
+    return new Promise(async (resolve, reject) => {
       try {
         const walletToOpen = this.getOne(name);
+        const net = network.getSelectedNetwork().net;
+
         const wif = wallet.decrypt(walletToOpen.encryptedWIF, passphrase);
         const account = new wallet.Account(wif);
+
+        let walletToOpenNew = storageNew.getOne(wif);
+        let mnemonic;
+        let encryptedMnemonicString;
+        let btcTestnetWalletClient;
+        let btcMainnetWalletClient;
+        let btcKey;
+
+        if (walletToOpen.encryptedMnemonicString === undefined) {
+          const dataBTCgen = await this.generateBTCETHWallet(name);
+          if (dataBTCgen.err) {
+            return reject('An error occured while trying to generate a new Bitcoin & Ethereum wallet.');
+          }
+          mnemonic = dataBTCgen.key.mnemonic;
+          btcKey = dataBTCgen.key;
+          btcTestnetWalletClient = dataBTCgen.btcTestnetWalletClient;
+          btcMainnetWalletClient = dataBTCgen.btcMainnetWalletClient;
+          const encryptedMnemonic = CryptoJS.AES.encrypt(mnemonic, account.WIF);
+          encryptedMnemonicString = encryptedMnemonic.toString();
+
+          this
+            .add(name, {
+              label: name,
+              encryptedWIF: walletToOpen.encryptedWIF,
+              address: walletToOpen.address,
+              scriptHash: walletToOpen.scriptHash,
+              encryptedMnemonicString,
+            })
+            .sync();
+
+          storageNew
+            .add(account.WIF, {
+              label: account.WIF,
+              encryptedMnemonicString,
+              isNew: true,
+            })
+            .sync();
+        } else {
+          const mnemonicBytes = CryptoJS.AES.decrypt(walletToOpen.encryptedMnemonicString, wif);
+          mnemonic = mnemonicBytes.toString(CryptoJS.enc.Utf8);
+
+          if (walletToOpenNew.isNew) {
```

Comment:
Should this be run also after line 125 if coming from the other branch? Should this be part of create a new wallet? When does this value ever get set to false?

Reply:
This will be true when creating a new wallet or come from other branches, otherwise, it will not create new wallet addresses.